.github/workflows/event-pull_request.yml

```diff
@@ -0,0 +1,58 @@
+name: Pull Request CI
+
+on:
+  pull_request:
+
+jobs:
+  check-if-docs-only:
+    uses: ./.github/workflows/task-check-docs.yml
+
+  basic-tests:
+    name: basic tests
+    runs-on: ubuntu-latest
+    needs: check-if-docs-only
+    if: needs.check-if-docs-only.outputs.only-docs-changed == 'false'
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+      cancel-in-progress: true
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+      - name: setup python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+      - name: install dependencies
+        run: sudo .install/install_script.sh
+      - name: install python packages
+        run: pip3 install -r requirements.txt
+      - name: check format
+        run: make check-format
+      - name: unit tests
+        run: make unit_test
+      - name: flow tests
+        run: make flow_test
+
+  coverage:
```

Comment:
So now coverage comes after the basic test job, but before it was in parallel, no? Any reason for making it sequential?

Reply:
No. It wasn't in parallel before. Only after the flow test passes, since we are running on an EC2 instance for coverage